scripts/run.py

```diff
@@ -40,19 +52,24 @@ def main(args):
     temp_dir = options.dump
     if not os.path.exists(temp_dir):
       os.makedirs(temp_dir)
-  s_file = os.path.join(temp_dir, basename + '.s')
-  wast_file = os.path.join(temp_dir, basename + '.wast')
-  wasm_file = os.path.join(temp_dir, basename + '.wasm')
 
   try:
-    subprocess.check_call([WACC, '-fno-builtin', options.c_file, '-o', s_file])
-    subprocess.check_call([sys.executable, WASMATE, s_file, '-o', wast_file])
-    subprocess.check_call([SEXPR_WASM, wast_file, '-o', wasm_file])
-    process = subprocess.Popen([D8, options.js_file, '--', wasm_file],
-                               stderr=subprocess.PIPE)
-    _, stderr = process.communicate()
-    if process.returncode != 0:
-      raise Error(stderr)
+    if options.js_env:
+      s_file = os.path.join(temp_dir, basename + '.s')
+      wast_file = os.path.join(temp_dir, basename + '.wast')
+      wasm_file = os.path.join(temp_dir, basename + '.wasm')
+
+      subprocess.check_call([WACC, '-fno-builtin', options.c_file, '-o', s_file, '-I', '.'])
+      subprocess.check_call([sys.executable, WASMATE, s_file, '-o', wast_file])
+      subprocess.check_call([SEXPR_WASM, wast_file, '-o', wasm_file])
+      call_and_suppress_stderr([D8, options.js_env, '--', wasm_file])
+
+    if options.native_env:
+      executable_file = os.path.join(temp_dir, basename)
+      subprocess.check_call([CLANG, '-O2', options.c_file, options.native_env,
+                             '-o', executable_file, '-I', '.'])
+      call_and_suppress_stderr([executable_file])
```

Comment:
maybe we'd want to keep stderr for the native file, since we don't expect stderr spam.